File: src/main/java/requestreplyapi/Registry/ExtendedEntry.java
package requestreplyapi.Registry;

public class ExtendedEntry extends Entry{
    private String entryType;

    public ExtendedEntry(String theDest, int thePort, String entryType) {
        super(theDest, thePort);
        this.entryType = entryType;
    }

    public String type(){
        return entryType;
    }

    @Override
    public String toString() {
        return "ExtendedEntry{" +
                "entryType='" + entryType + '\'' +
                super.toString() +
                '}';
    }
}

File: ObjectRequestBrokerCS/ORB/requestreplyapi/entries/ExtendedEntry.cs
namespace ORB.requestreplyapi.entries
{
	public class ExtendedEntry : Entry
	{
		private string entryType;

		public ExtendedEntry(string theDest, int thePort, string entryType) : base(theDest, thePort)
		{
			this.entryType = entryType;
		}

		public ExtendedEntry()
		{
		}

		public virtual string Type()
		{
			return entryType;
		}

		public override string ToString()
		{
			return "ExtendedEntry{" +
					"entryType='" + entryType + '\'' +
					base.ToString() + '}';
		}
	}

}